auth.go

```diff
@@ -283,3 +284,101 @@ func (c *Client) AuthCheck(conf *AuthConfiguration) (AuthStatus, error) {
 	}
 	return authStatus, nil
 }
+
+// helperCredentials represents credentials commit from an helper
+type helperCredentials struct {
+	Username string `json:"Username,omitempty"`
+	Secret   string `json:"Secret,omitempty"`
+}
+
+// NewAuthConfigurationsFromCredsHelpers returns AuthConfigurations from
+// installed credentials helpers
+func NewAuthConfigurationsFromCredsHelpers(registry string) (*AuthConfiguration, error) {
+	// Load docker configuration file in order to find a possible helper provider
+	pathsToTry := cfgPaths(os.Getenv("DOCKER_CONFIG"), os.Getenv("HOME"))
```

Comment:
Shouldn't this be part of `NewAuthConfigurationsFromDockerCfg`?

Also, can you add a test?

Reply:
(for the test, feel free to add a script to testdata and invoke that, it's easier than mocking exec.Command)

Reply:
At first, I wanted to put as a second way of authenticating in `NewAuthConfigurationsFromDockerCfg` but the function name clearly state it comes from docker config. Which not really what I'm doing. Thats why I made this like this. 

I have no problem doing it though. It's just me overthinking it :)

I will add a test.

Reply:
After some thinking, I'll need your opinion.

I also did it because I need to pass the registry name in order to feed it to the found helper. If I put it in `NewAuthConfigurationsFromDockerCfg`, it would change the function way (thinking about backward compatibility).

What do you think?

Reply:
Sorry for pushing but could you give me your opinion about my last comment ? Thanks

Reply:
Hey @teintuc, sorry for taking so long to get back to you. Things have been crazy over here x)

> After some thinking, I'll need your opinion.
> 
> I also did it because I need to pass the registry name in order to feed it to the found helper. If I put it in `NewAuthConfigurationsFromDockerCfg`, it would change the function way (thinking about backward compatibility).
> 
> What do you think?

This is a good point that I didn't realize in my original message. We can keep this version, as we need to take the registry. Let me just give this another look before merging! 

Thank you very much for your contribution and patience!